cypress-io/cypress
# fix: fix browser focus issue on MacOS

- Closes https://github.com/cypress-io/cypress/issues/21743
### User facing changelog


Fix a MacOS specific bug where the browser would not be correctly focused when launched.

### Additional details


For reasons unknown, if you quickly jiggle your mouse back and forth when launching Chrome or Firefox browsers on MacOS, the browser would not be correctly focused. You can see this below in the first video - I jiggle my mouse, and then I need to click the search bar a bunch of times to get the browser to properly focus.

**Note** this also happens in Cypress 9 on the latest MacOS version. I reproduced it here: https://github.com/cypress-io/cypress/issues/21743#issuecomment-1331598390

There was a work around which was to switch to any other window, then switch back, and it would work as expected. I don't know why this fixes the issue, but i does. I was able to do the same thing programmatically, but simply showing and immediately hiding the electron app in code.

## Bug


https://user-images.githubusercontent.com/19196536/204693368-d3e96843-8ae1-45fb-bc82-02f9513d1f14.mov


## Fix


https://user-images.githubusercontent.com/19196536/204693495-ff657962-15f4-4530-a648-a5723ffc3fca.mov


### Steps to test


Note: You need to use MacOS to test this. 

1. Reproduce the issue (explained above).
2. Get this branch.
3. Try to reproduce it - you shouldn't be able to.

I see no practical way to write an automated test around this behavior. 🤔 

### How has the user experience changed?


The launched browser is now correctly focused.

### PR Tasks


- [na] Have tests been added/updated?
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
